annot/src/main/java/com/predic8/membrane/annot/generator/kubernetes/K8sJsonSchemaGenerator.java

```diff
@@ -13,29 +13,25 @@
    limitations under the License. */
 package com.predic8.membrane.annot.generator.kubernetes;
 
-import com.predic8.membrane.annot.ProcessingException;
-import com.predic8.membrane.annot.generator.kubernetes.model.ISchema;
-import com.predic8.membrane.annot.generator.kubernetes.model.Schema;
-import com.predic8.membrane.annot.generator.kubernetes.model.SchemaObject;
+import com.predic8.membrane.annot.*;
+import com.predic8.membrane.annot.generator.kubernetes.model.*;
 import com.predic8.membrane.annot.model.*;
 
-import javax.annotation.processing.ProcessingEnvironment;
-import javax.tools.FileObject;
-import java.io.BufferedWriter;
-import java.io.IOException;
-import java.io.Writer;
+import javax.annotation.processing.*;
+import javax.tools.*;
+import java.io.*;
 import java.util.*;
 
 /**
- * Generates Json Schema draft 4 to validate kubernetetes CustomResourceDefinitions.
+ * Generates JSON Schema (draft 2019-09/2020-12) to validate Kubernetes CustomResourceDefinitions.
```

Comment:
Kubernetes still uses OpenAPI 3.0 (There is a [plan](https://github.com/kubernetes/kubernetes/issues/134684) to move to 3.1.). OpenAPI 3.0 uses JSON Schema `00`. While we declared a `04` schema, all features used were backwards-compatible with `00`. Kubernetes has the additional restriction that you cannot define types (=`$defs` cannot be used).

Therefore, the simplest fix is probably to leave `com.predic8.membrane.annot.generator.kubernetes` as-is.